### PR TITLE
Eliminate repetition in the construction of `AnalyzerRun`s

### PR DESCRIPTION
**Pull request by fviernau**
A non-function refactoring which first eliminates some obsolete constants for default values and
then makes use of `AnalyzerRun.EMPY.copy()` to eliminate quite a bit of repetition.

In context of #5681.

### CODE REVIEW

File: model/src/test/kotlin/config/ExcludesTest.kt [hunk 1]
```diff
@@ -73,7 +73,7 @@ class ExcludesTest : WordSpec() {
             repository = Repository.EMPTY,
             analyzer = AnalyzerRun(
                 environment = Environment(),
-                config = AnalyzerConfiguration(allowDynamicVersions = false),
+                config = AnalyzerConfiguration(),
```
Comment (sschuberth) [inline model/src/test/kotlin/config/ExcludesTest.kt]: There is one more place in `RequirementsCommand`, mind simplifying that too as part of this commit?

File: reporter/src/funTest/kotlin/reporters/spdx/SpdxDocumentReporterFunTest.kt [hunk 2]
```diff
@@ -168,7 +168,7 @@ private val ortResult = OrtResult(
     ),
     analyzer = AnalyzerRun(
         environment = Environment(),
-        config = AnalyzerConfiguration(allowDynamicVersions = true),
+        config = AnalyzerConfiguration(),
```
Comment (sschuberth) [inline reporter/src/funTest/kotlin/reporters/spdx/SpdxDocumentReporterFunTest.kt]: Mind squashing this to the previous commit? I believe it's easier to review the same change in a single blow, and it avoids you to retype the same commit message.

File: model/src/test/kotlin/licenses/TestData.kt [hunk 3]
```diff
@@ -180,7 +180,7 @@ val ortResult = OrtResult(
     ),
     analyzer = AnalyzerRun(
         environment = Environment(),
-        config = AnalyzerConfiguration(allowDynamicVersions = true),
+        config = AnalyzerConfiguration(),
```
Comment (sschuberth) [inline model/src/test/kotlin/licenses/TestData.kt]: Same here, I'd prefer a squash.

File: advisor/src/test/kotlin/AdvisorTest.kt [hunk 4]
```diff
@@ -132,9 +130,7 @@ private fun createAdvisor(providers: List<AdviceProvider>): Advisor {
  */
 private fun createOrtResultWithPackages(packages: List<Package>): OrtResult =
     OrtResult.EMPTY.copy(
-        analyzer = AnalyzerRun(
-            environment = Environment(),
-            config = AnalyzerConfiguration(),
+        analyzer = AnalyzerRun.EMPTY.copy(
```
Comment (sschuberth) [inline advisor/src/test/kotlin/AdvisorTest.kt:133]: Nice changes!

File: model/src/main/kotlin/AnalyzerRun.kt [hunk 5]
```diff
@@ -29,14 +29,14 @@ import org.ossreviewtoolkit.utils.ort.Environment
...
+     * The [Instant] the analyzer was started..
      */
-    val startTime: Instant = Instant.EPOCH,
+    val startTime: Instant,
 
     /**
      * The [Instant] the analyzer has finished. The default value exists only for backward compatibility.
      */
-    val endTime: Instant = Instant.EPOCH,
+    val endTime: Instant,
```
Comment (sschuberth) [inline model/src/main/kotlin/AnalyzerRun.kt:39]: The comment above should also be changed here.

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Review by sschuberth — DISMISSED** [review]

**Comment by sschuberth** [comment]
Unfortunately, there's a test failure in "AnalyzerRun without timestamps can be deserialized".

**Comment by fviernau** [comment]
> Unfortunately, there's a test failure in "AnalyzerRun without timestamps can be deserialized".

I've dropped the test. Hope the other tests pass now.

**Comment by sschuberth** [comment]
> I've dropped the test. Hope the other tests pass now.

Yes, but now there are a few unused imports.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**